pokemonred_puffer/environment.py

```diff
@@ -226,6 +211,25 @@ def __init__(self, env_config: pufferlib.namespace):
             RedGymEnv.env_id.buf[2] = (env_id >> 8) & 0xFF
             RedGymEnv.env_id.buf[3] = (env_id) & 0xFF
 
+    @functools.cached_property
```

Comment:
Curious, what is the motivation for moving this to a cached property? Maybe it should be a class-level property?

Reply:
I know it's a style issue, and I'm more familiar with this from nmmo. I thought of two reasons while making a example of customized obs -- https://github.com/kywch/poke-trainer/blob/template/environment.py#L30

1. Separating obs related stuff from init (and other pyboy stuff) can signal people that they can easily play with it too
2. Making it read-only can prevent mistakes (yeah, it might be rare, but ...)

Reply:
I'll keep this. But it just feels a [class variable](https://www.digitalocean.com/community/tutorials/understanding-class-and-instance-variables-in-python-3#class-variables) would make more sense. If a user wants to they could always create a new class that inherits from RedGymEnv and set it to something else.

The one defense I see in this setup is if the observation is configurable.

Reply:
I reverted these back to class vars. I forgot about the existing baselines ;)